Flabbergasted-Flamingos/bangazon-commerce-app
# Rb Customer Model

# Description
The customer model inherits from User, so in DB browser after you register a new user that user should be in website.user field as well as the user table. The user_id functions as a foreign key on the customer instance.

## Number of Fixes
Fixes our customer model so that it inherits from user

## Related Ticket(s)
several tickets

## Problem to Solve
This solves the problem of adding location to our customer's and sellers without being able to access the user model directly

## Proposed Changes
This affects the views (register), models(customer), forms and register.template

## Expected Behavior
Now when a user register's there is an added field of "location" This is registered in the customer table, but everything else is stored in the user built in django model.

## Steps to Test Solution

1. 
```
git fetch --all
git checkout rb_landing#15
rm -rf website/migrations
rm db.sqlite3
python manage.py makemigrations website
python manage.py migrate
python manage.py runserver
```
2. After running server go to the 'register' url.
Add a new user
after pressing submit it should take you to a page that says welcome to Bangazon and that username
Then, open DB browser and make sure there is an instance of that user in the user table as well as the website.customer table.


## Documentation
[ x] I added documentation on the customer method that is a link to the tutorial I used for connecting these two things
